modules/core/src/main/java/org/apache/ignite/transactions/Transaction.java

```diff
@@ -286,4 +299,53 @@
      * @throws IgniteException If suspension failed.
      */
     public void suspend() throws IgniteException;
-}
\ No newline at end of file
+
+    /**
+     * Use this method to identify a point in a transaction to which you can later roll back.
+     * <p>
+     * Savepoint names must be distinct within a given transaction.
+     * If you create a second savepoint with the same identifier as an earlier savepoint,
+     * then the earlier savepoint is erased.
+     * <p>
+     * After a savepoint has been created, you can either continue processing,
+     * commit your work, roll back the entire transaction, or roll back to the savepoint.
+     *
+     * @param name savepoint ID.
+     * @throws IllegalArgumentException If savepoint with such name already exists.
+     */
+    public void savepoint(String name) throws IllegalArgumentException;
```

Comment:
"If you create a second savepoint with the same identifier as an earlier savepoint then **the earlier savepoint is erased**"
"@**throws IllegalArgumentException** If savepoint with such name already exists."
Its either bug in javadoc or weird behavior. If you throw exception, then you must not erase old savepoint.
If you erase - you must not throw exception